### PR TITLE
Display processing status in a separate HTML element

### LINKED ISSUE #50: Stop button bugs
First of all, I don't think that **stop** button should even be present on the startup screen. If possible it should be added after task submit or at least do nothing when pressed and processing hasn't been started yet.

Then, in the latest version stop button doesn't turn red when processing is in progress.

### LINKED ISSUE #51: Do a better job at displaying processing status
Do a better job at displaying processing status (in progress, stopped, finished).
Currently, the way to know if processing has been finished is to either scroll to the bottom of the page or look to the log box, but this is not as intuitive as I want it to be.
I think it would be a great idea to display processing status near time elapsed clock.

Here is how it looks now

### PR DESCRIPTION
**Pull request by No-name-and-surname**
Closes #51, closes #50

### CODE REVIEW

File: static/index.html [hunk 1]
```diff
@@ -42,6 +42,7 @@
         </div>
 
         <p id="timer"></p>
+        <p id="init_info_1"></p>
```
Comment (pavtiger) [inline static/index.html:45]: init_info_1 is such a not explanatory name. I think you should name it somehow different to be easier to understand what is it

File: static/main.js [hunk 2]
```diff
@@ -65,40 +64,39 @@ socket.on("nothing_found_on_page", function(message) {
...
-
-        for (let i = 0; i < header.length; i++) {
-            div.appendChild(header[i])
-        }
-
-        console_div.append(div);
-
-        let target_dropdown = div.querySelector(".main_body");
-        target_dropdown.style.display = "none";
+            let header = htmlToElements('<div class="row header align-items-center border" style="height: 10%">\n' +
```
Comment (pavtiger) [inline static/main.js:67]: How did this happen? Indentation is wrong here, double indentation in for loop. This whole paragraph just moved one indentation to the right. I ask you one more time to check git diff to make sure that things like this do not get commited

File: static/main.js [hunk 3]
```diff
@@ -182,7 +180,13 @@ socket.on("work_finish", function(message) {  // Receive and download results
             document.body.removeChild(a);
         });
     }
+    if (processing_in_progress) {
+        let init_info_1 = document.getElementById("init_info_1");
+        init_info_1.innerHTML = "";
```
Comment (pavtiger) [inline static/main.js:185]: Using = operand overwrites current value, why do you have to equal it to "" if next line you equal it to something else

File: static/main.js [hunk 4]
```diff
@@ -182,7 +180,13 @@ socket.on("work_finish", function(message) {  // Receive and download results
...
+        init_info_1.innerHTML = '<div style="color:green;font-size:50px;font-family: terminal, sans-serif;justify-content: center;align-items: center;display: flex;">Processing finished</div>'
+    }
     processing_in_progress = false;
+
```
Comment (pavtiger) [inline static/main.js:189]: Just an empty line at the end of a function separating nothing?

File: static/main.js [hunk 5]
```diff
@@ -222,20 +226,28 @@ form.addEventListener("submit", (e) => {  // Submit button press event
...
-
-    let stop_button = document.getElementById("stop_button");
-    stop_button.style.color = "black";
-    processing_in_progress = false;
-
-    socket.emit("stop");
-
     let console_element = document.getElementById("init_info");
-    console_element.innerHTML += "Processing stopped\n";
+    if(processing_in_progress) {
```
Comment (pavtiger) [inline static/main.js:236]: The structure for loops is
```python
if () {

}
```
There is *always* a space after `if`

File: static/main.js [hunk 6]
```diff
@@ -222,20 +226,28 @@ form.addEventListener("submit", (e) => {  // Submit button press event
...
+        socket.emit("stop");
+        let stop_button = document.getElementById("stop_button");
+        stop_button.style.color = "black";
+        processing_in_progress = false;
+    } else {
+        let init_info_1 = document.getElementById("init_info_1");
+        init_info_1.innerHTML = "";
+        console_element.innerHTML = '<div style="color:red;font-size:50px;">Error: you haven`t started yet';
+        processing_in_progress = false;
+
```
Comment (pavtiger) [inline static/main.js:249]: empty line again?

**Review by pavtiger — CHANGES_REQUESTED** [review]
Well, quite a few things to fix, but we're learning. One global suggestion I would like you to implement is to set timer after processing status. Because the main thing in focus should be the status, timer is just an additional information. So move it below status and make it a bit smaller (font) as well. Thanks!

File: recognise.py [hunk 7]
```diff
@@ -286,7 +284,6 @@ async def process_by_link(link, quality, limit, sid, download_on_finish):
         return False
 
     # Main spreadsheet processing
-    await emit_message('Processing started', sid)
```
[the remark this answers is not in the document]
Comment (DashaD2205) [reply recognise.py:289]: I just replace it by another element :)

File: static/index.html (hunk 1 above)
Comment (DashaD2205) [reply static/index.html:45]: What name would be better for this element?
I can't realise that, because i think it's okey

Comment (pavtiger) [reply static/index.html:45]: Well in no world (almost) is naming a variable `something_1` a good idea. As a person who tries to understand what is happening, I cannot understand why is it created, what is stands for.
So a better name would be `processing_status` as `init_info` was originally just a console from server that printed out initialization information and current steps

File: static/main.js [hunk 8]
```diff
@@ -211,8 +218,8 @@ form.addEventListener("submit", (e) => {  // Submit button press event
     console_element.innerHTML = "";
 
     // Clear text field
-    let init_info = document.getElementById("init_info");
-    init_info.innerHTML = "";
+    let init_info_1 = document.getElementById("init_info_1");
+    init_info_1.innerHTML = "";
```
Comment (pavtiger) [inline static/main.js:222]: Why do you clear innerHTML if on line 232 you assign this element a new value with "Processing started"

File: static/main.js [hunk 9]
```diff
@@ -222,20 +229,27 @@ form.addEventListener("submit", (e) => {  // Submit button press event
...
+        init_info_1.innerHTML = "";
+        console_element.innerHTML = '<div style="color:red;font-size:50px;">Processing stopped</div>';
+        socket.emit("stop");
+        let stop_button = document.getElementById("stop_button");
+        stop_button.style.color = "black";
+        processing_in_progress = false;
+    } else {
+        let init_info_1 = document.getElementById("init_info_1");
+        init_info_1.innerHTML = "";
+        console_element.innerHTML = '<div style="color:red;font-size:50px;">Error: you haven`t started yet';
```
Comment (pavtiger) [inline static/main.js:250]: On lines [210](https://github.com/pavtiger/Parse-tables-from-PDF/pull/69/files#diff-acd2228669a20570c1c6bfe003b0b075ded45f1270f2dac8fda906c9583f4531R210) and [202](https://github.com/pavtiger/Parse-tables-from-PDF/pull/69/files#diff-acd2228669a20570c1c6bfe003b0b075ded45f1270f2dac8fda906c9583f4531R202) you changed this message in processing_status box to alert() but here you do the other thing. So to maintain consistency you have to choose one and do all around the code. If you want to do alerts, then do an alert here as well

File: static/main.js [hunk 10]
```diff
@@ -222,20 +229,27 @@ form.addEventListener("submit", (e) => {  // Submit button press event
...
+        processing_in_progress = false;
```
Comment (pavtiger) [inline static/main.js:251]: Unnecessary. You are in an if branch where `processing_in_progress` is false, so you don't need to set it to false again

**Review by pavtiger — CHANGES_REQUESTED** [review]
Much better, still you have a few unfixed issues and the thing I asked in previous review to swap processing status and timer. Please do that

File: static/main.js (hunk 9 above)
Comment (DashaD2205) [reply static/main.js]: okay, got it.

**Comment by pavtiger** [comment]
Due to your complete lack of attentiveness and competence you force pushed and overrode one of my commits with your force push. I once again ask you to be more attentive with your contributions

**Review by pavtiger — APPROVED** [review]
Finally! This was very tough but we made it through. Just please be more careful and attentive because shear amount of mistakes is shocking

**Pull request merged**